geocoder/base.py

```diff
@@ -713,6 +725,18 @@ def __repr__(self):
         else:
             return base_repr.format(u'#%s results' % len(self))
 
+    def _build_headers(self, provider_key, **kwargs):
```

Comment:
yep, a bit like "hooks" that allows to factorize the main part of the code in base.py, and offers at the same time high flexibility in the cutomization 👍